.github/workflows/haskell.yml

```diff
@@ -465,3 +465,34 @@ jobs:
         env:
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
           SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
+
+  SRP-validity:
+    name: Check if formal-ledger-specifications SRP is valid
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Extract commit hash from formal-ledger-specifications SRP
+      run: |
+        RES=$(sed -ne \
+          '/ *location: *https:\/\/github.com\/IntersectMBO\/formal-ledger-specifications.git/,/--sha256/ p' \
+          cabal.project \
+        | sed -ne 's/^ *tag: *//p')
+        echo "TAG=$RES" >> $GITHUB_ENV
```

Comment:
Once you no longer need the second `checkout` action, I don't think you need to make these separate GHA steps. You could change `RES` to `TAG` and then use it lower down.